src/simulation/onboardDataHandling/storageUnit/partitionedStorageUnit.cpp

```diff
@@ -58,7 +58,20 @@ void PartitionedStorageUnit::customReset(uint64_t currentClock){
 void PartitionedStorageUnit::addPartition(std::string dataName){
     dataInstance tmpDataInstance;
     strncpy(tmpDataInstance.dataInstanceName, dataName.c_str(), sizeof(tmpDataInstance.dataInstanceName));
-    tmpDataInstance.dataInstanceSum = 0.0;
+    tmpDataInstance.dataInstanceSum = 0;
     this->storedData.push_back(tmpDataInstance);
     return;
+}
+
+/*! Adds a specific amount of data to the specified partitions once
+ @param partitionNames  //Vector of partition names
+ @param data            //Vector of data to be added to each partition in partitionNames
+ @return void
+ */
+void PartitionedStorageUnit::setDataBuffer(std::vector<std::string> partitionNames, std::vector<long long int> data){
```

Comment:
This should be `int64_t` also, (assuming I'm not missing something here...I could be :) )